graylog2-server/src/main/java/org/graylog/plugins/views/search/engine/QueryBackend.java

```diff
@@ -137,9 +139,16 @@ default ExplainResults.QueryExplainResult doExplain(SearchJob job, Query query,
         final ImmutableMap.Builder<String, ExplainResults.ExplainResult> builder = ImmutableMap.builder();
 
         query.searchTypes().forEach(s -> {
-            final Set<ExplainResults.IndexRangeResult> indicesForQuery = indexRangesForStreamsInTimeRange(
-                    query.effectiveStreams(s), query.effectiveTimeRange(s))
-                    .stream().map(ExplainResults.IndexRangeResult::fromIndexRange).collect(Collectors.toSet());
+            final Set<String> streamNames = query.effectiveStreams(s);
+            final Set<String> streamTitles = streamNames.stream()
+                    .map(this::streamTitle)
+                    .flatMap(Optional::stream)
```

Comment:
maybe we should filter out `null` values here?

Reply:
good point- fixed